pkgs/development/libraries/nanodbc/default.nix

```
@@ -0,0 +1,29 @@
+{ lib, stdenv, fetchFromGitHub, cmake, unixODBC }:
+
+stdenv.mkDerivation rec {
+  pname = "nanodbc";
+  version = "2.13.0";
+
+  src = fetchFromGitHub {
+    owner = "nanodbc";
+    repo = "nanodbc";
+    rev = "v${version}";
+    sha256 = "1q80p7yv9mcl4hyvnvcjdr70y8nc940ypf368lp97vpqn5yckkgm";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ unixODBC ];
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DBUILD_STATIC_LIBS=ON"
+  ];
```

We'd want to build the shared lib only by default.
And the static lib when requested.

```suggestion
  cmakeFlags = 
    lib.optionals (!stdenv.hostPlatform.isStatic) [ "-DBUILD_SHARED_LIBS=ON" ]
    ++ lib.optionals (stdenv.hostPlatform.isStatic) [ "-DBUILD_STATIC_LIBS=ON" ];
```

---

```suggestion
  cmakeFlags = if (stdenv.hostPlatform.isStatic) then
    [ "-DBUILD_STATIC_LIBS=ON" ]
  else
   [ "-DBUILD_SHARED_LIBS=ON" ];
```